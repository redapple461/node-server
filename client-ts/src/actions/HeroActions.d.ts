export declare const SET_MARVEL = "SET_MARVEL";
export declare const SET_DC = "SET_DC";
export declare const SET_BOTH = "SET_BOTH";
export declare const GET_DATA = "GET_DATA";
export declare const HEROES_LOADED = "HEROES_LOADED";
export declare const ADD_HERO = "ADD_HERO";
export declare const DELETE_HERO = "DELETE_HERO";
export declare const INIT_HERO = "INIT_HERO";
export declare const UPDATE_NAME = "UPDATE_NAME";
export declare const UPDATE_ADD_HERO_NAME = "UPDATE_ADD_HERO_NAME";
export declare const UPDATE_ADD_HERO_UNIVERSE = "UPDATE_ADD_HERO_UNIVERSE";
export declare const CLEAR_DETAIL_HERO = "CLEAR_DETAIL_HERO";
export declare const DETAIL_LOADED = "DETAIL_LOADED";
export declare const UPDATE_HERO = "UPDATE_HERO";
export declare const UPDATE_UNIVERSE = "UPDATE_UNIVERSE";
export declare const ADD_ASKILL = "ADD_ASKILL";
export declare const REMOVE_ASKILL = "REMOVE_ASKILL";
export declare const CLEAR_ADD_HERO = "CLEAR_ADD_HERO";
export declare const SET_EMAIL = "SET_EMAIL";
export declare const SET_PASSWORD = "SET_PASSWORD";
export declare const SET_REG_EMAIL = "SET_REG_EMAIL";
export declare const SET_REG_PASSWORD = "SET_REG_PASSWORD";
export declare const SET_USER_ID = "SET_USER_ID";
export declare const SET_JWT = "SET_JWT";
export declare const LOGOUT = "LOGOUT";
export declare const SET_USER_NAME = "SET_USER_NAME";
export declare const SET_USER_SURNAME = "SET_USER_SURNAME";
export declare const SET_USER_PHONE = "SET_USER_PHONE";
export declare const CLEAR_REG_DATA = "CLEAR_REG_DATA";
