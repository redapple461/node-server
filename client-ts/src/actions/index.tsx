import { Hero } from '../models/Hero';
import {HeroActionsType} from '../actions/types/HeroTypes';
import * as actions from '../actions/HeroActions';

export function setMarvel (): HeroActionsType {
	return{
		type: actions.SET_MARVEL
	};
}

export  function setDC (): HeroActionsType {
	return{
		type: actions.SET_DC
	};
}

export function setBoth (): HeroActionsType{
	return{
		type: actions.SET_BOTH
	};
}

export function getData (data: Hero[]): HeroActionsType {
	return{
		type: actions.GET_DATA,
		data: data
	};
}

export function loadComplete(): HeroActionsType {
	return{
		type: actions.HEROES_LOADED
	};
}

export function addHero(hero: Hero): HeroActionsType {
	return{
		type: actions.ADD_HERO,
		hero: hero
	};
}

export function deleteHero(id: number): HeroActionsType{
	return{
		type: actions.DELETE_HERO,
		id: id
	};
}

export function initDetailHero(hero: Hero): HeroActionsType {
	return{
		type: actions.INIT_HERO,
		hero: hero
	};
}

export function updateName(name: string): HeroActionsType {
	return{
		type: actions.UPDATE_NAME,
		newName: name
	};
}

export function updateUniverse(universe: string): HeroActionsType{
	return{
		type: actions.UPDATE_UNIVERSE,
		newUniverse: universe
	};
}

export function updateHero(): HeroActionsType {
	return{
		type: actions.UPDATE_HERO
	};
}

export function updateAddHeroName(name: string): HeroActionsType{
	return{
		type: actions.UPDATE_ADD_HERO_NAME,
		addName: name,
	};
}

export function updateAddHeroUniverse(universe: string): HeroActionsType{
	return{
		type: actions.UPDATE_ADD_HERO_UNIVERSE,
		addUniverse: universe
	};
}

export function clearDetailHero(): HeroActionsType{
	return{
		type: actions.CLEAR_DETAIL_HERO
	};
}

export function isLoad(): HeroActionsType {
	return{
		type: actions.DETAIL_LOADED
	};
}
