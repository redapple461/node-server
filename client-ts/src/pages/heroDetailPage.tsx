import React, { useEffect } from 'react';
import './css/heroDetail.css';
import { useSelector, useDispatch } from 'react-redux';
import * as actions from '../actions';
import { NoPage } from '../components/404';
import {  Details } from '../components/detailPage';
import { RadioButton } from '../components/radio';
import { Button } from '../components/button';
import { getByName } from '../http/httpHook';
import { HeroStore } from '../interfaces/iStore/HeroStore';
import { updateHero } from '../http/httpHook';

export const HeroDetailPage = (props: any) => {
	const token = useSelector((state: HeroStore) => state.jwt);
	const oldname = useSelector((state: HeroStore) => state.oldName);
	const detailHero = useSelector((state: HeroStore) => state.detailHero);
	const dispatch = useDispatch();
	const isLoad = useSelector((state: HeroStore) => state.isLoad);
	async function fetchData () {
		try {
			await getByName(props.match.params.name, token)
			.then(res => {
				if (res.message) {
					return window.M.toast({html: 'No auth!!!'});
				}
				dispatch(actions.initDetailHero(res));
		});
		} catch (e) {
			// tslint:disable-next-line: no-console
			console.log(e);
		}
		}
	const goBack = () => {
		props.history.goBack();
	};

	useEffect(() => {
			if (isLoad === false) {
				fetchData();
				dispatch(actions.isLoad());
			}
	});

	if (!detailHero) {
		return (
			<NoPage name={props.match.params.name} />
		);
	}
	const logout = () => {
		dispatch(actions.logout());
		localStorage.removeItem('userData');
	};
	//               onChange={(e) => dispatch(actions.updateUniverse(e.target.value))}
	return(
		<>
				<Button className='waves-effect waves-light btn rightbtn' text='Logout' onClick={() => logout()}/>
				<Details name={detailHero.name} id={detailHero.id} universe={detailHero.universe} skills={detailHero.skills}/>
				<div className='form'>
					<p> Type new name of hero and choose his universe</p>
					<input
						type='text'
						placeholder={detailHero.name}
						onChange={(e) => {dispatch(actions.updateName(e.target.value)); }}
					/>
					<div>
						<RadioButton
								className='with-gap'
								value='Marvel'
								text='Marvel'
								checked={detailHero.universe === 'Marvel'}
								dispatch={() =>  dispatch(actions.updateUniverse('Marvel'))}
							/>
						<RadioButton
							className='with-gap'
							value='DC'
							text='DC'
							checked={detailHero.universe === 'DC'}
							dispatch={() =>  dispatch(actions.updateUniverse('DC'))}
						/>
					</div>
				</div>

				<Button
					className='waves-effect waves-light btn'
					type='button'
					onClick={() => {goBack(); }}
					text='Back'
				/>
				<Button
					className='waves-effect waves-light btn'
					type='button'
					onClick={() => { updateHero(oldname, detailHero, token).then(() => dispatch(actions.updateHero()))} }
					text='Save'
				/>
			</>
	);
};
