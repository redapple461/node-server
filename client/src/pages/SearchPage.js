import React from 'react'

export class SearchPage extends React.Component{
    constructor(props){
        super(props);
        console.log(this.props.state);
    }
    render(){
        return(
            <div>

            </div>
        )
    }
}