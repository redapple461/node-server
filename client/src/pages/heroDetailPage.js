import React, { useEffect } from 'react'
import './css/heroDetail.css'
import { useSelector, useDispatch } from 'react-redux'
import * as actions from '../actions'
import { NoPage } from '../components/404'
import {  Details } from '../components/detailPage'
import { RadioButton } from '../components/radio'
import { Button } from '../components/button'



export const HeroDetailPage = (props) => {
   
    const detailHero = useSelector(state => state.detailHero)
    const dispatch = useDispatch();
    const isLoad = useSelector(state => state.isLoad)
    async function fetchData() {
        try{
            await fetch('http://localhost:4000/getHero/'+props.match.params.name)
            .then(res => res.json())
            .then(res => dispatch(actions.initDetailHero(res[0])));
        }catch(e){
            
        }
      }
    const goBack = () => {
   
        props.history.goBack()

      
    }  

    useEffect(()=>{
            if(!isLoad){
                fetchData()
                dispatch(actions.isLoad())
            }
    })


    if(!detailHero){
        return (
            <NoPage name = {props.match.params.name} />
        )
    }
    //               onChange={(e) => dispatch(actions.updateUniverse(e.target.value))}
    return(
        <div>

                <Details name = {detailHero.name} id={detailHero.id} universe={detailHero.universe}/>
                <div className="form">
                    <p> Type new name of hero and choose his universe</p>
                    <input
                     type="text" 
                     placeholder={detailHero.name}
                     onChange={(e) => {dispatch(actions.updateName(e.target.value))}}
                    ></input>
                    <div>
                        <RadioButton 
                                className ="with-gap"
                                value="Marvel"
                                text = "Marvel"
                                checked = {detailHero.universe === "Marvel"}
                                dispatch = { () =>  dispatch(actions.updateUniverse("Marvel"))}
                            />
                        <RadioButton 
                            className ="with-gap"
                            value="DC"
                            text = "DC"
                            checked = {detailHero.universe === "DC"}
                            dispatch = { () =>  dispatch(actions.updateUniverse("DC"))}
                        />
                    </div>
                </div>
                
                <Button
                    className="waves-effect waves-light btn" type="button" 
                    onClick={() => {goBack()} }
                    text = "Back"
                 />
                <Button 
                    className="waves-effect waves-light btn" 
                    type="button" 
                    onClick={() => {dispatch(actions.updateHero())}}
                    text = "Save"
                 />
            </div>
    )
}