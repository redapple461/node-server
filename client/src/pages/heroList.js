import React from 'react'
import {Link} from 'react-router-dom'
import {Heroes} from '../heroes'
import './css/heroList.css'
import { queryHelpers } from '@testing-library/react'
class HeroList extends React.Component{
    render(){
        return(
            <div>
                <h1> Heroes List</h1>
                <Link to="/main"> <strong>Dashborad</strong></Link>
                <ul>
                    {Heroes.map((hero)=>{
                        return <Link to={"/detailHero/"+hero.id}><li key={hero.id}>{hero.id}: {hero.name}</li></Link>
                    })}
                </ul>
            </div>
        )
    }
}

export default HeroList;