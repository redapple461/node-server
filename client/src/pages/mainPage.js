import React from 'react'
import {Link} from 'react-router-dom'
import './css/mainPage.css'


export  class MainPage extends React.Component {
    constructor(props){
        super(props);
        this.state = {universe: "",lastHeroes: []};
      //  this.getHeroes();
    }
    
    getHeroes(){
        fetch("http://localhost:3000/getHeroes")
            .then(res => res.json())
            .then(heroes => this.setState({lastHeroes: heroes.slice(heroes.length-4,heroes.length).reverse()}))
       // window.M.toast({html: "Data is fetched"});
    }
    componentDidMount(){
        this.getHeroes();        
    }
    componentDidUpdate(){
        this.getHeroes();    
    }
    onRadioChange = (e) => {
        this.setState({
            universe: e.target.value
        });
        if(e.target.value){
            window.M.toast({html: "Heroes will be displayed from : "+e.target.value+" universe"});
        }else{
            window.M.toast({html: "Heroes will be displayed from both universes"});
        }
    }
    onInputChange = (e) => {
        this.setState({
            nameToSearch: e.target.value
        });
    }

    render(){
    return (
       <div>
            <div>
                <h1> Tours of heroes</h1>
            </div>
            <p> </p>
            <div>
              <Link to ={
                  { 
                   pathname: "/heroes",
                   state: {universe: this.state.universe} 
                  }
                  }><button className="waves-effect waves-light btn" >Heroes</button></Link>
              <div >
                <label>
                        <input 
                            className="with-gap" 
                            name="group1" type="radio" 
                            value="Marvel" 
                            checked={this.state.universe === "Marvel"}
                            onChange={this.onRadioChange}
                        />
                        <span>Marvel</span>
                </label>
                <label>
                        <input 
                            className="with-gap"
                            name="group1" 
                            type="radio"
                            value="DC"
                            checked={this.state.universe === "DC"}
                            onChange={this.onRadioChange}
                        />
                        <span>DC</span>
                </label>
                <label>
                        <input
                            className="with-gap"
                            name="group1"
                            type="radio"
                            value="" 
                            checked={this.state.universe === ""}
                            onChange={this.onRadioChange}
                        />
                        <span>Both</span>
                </label>
              </div>
              <ul>
                    {this.state.lastHeroes.map((hero)=>{
                        return  <div className="dashboard" key={hero.id}>
                                    <Link  to={"/detailHero/"+hero.name} >
                                        <button className="waves-effect teal lighten-1 btn dashboard_btn">
                                            {hero.name}
                                        </button>
                                    </Link>
                                </div>
                    })}
              </ul>
            </div>
     
       </div>
    );
    }
}