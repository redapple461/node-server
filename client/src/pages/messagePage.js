import React from 'react'

export class MessagePage extends React.Component{
    render(){
        return(
            <div>
                <h2> 
Message service will be added along with all the functionality </h2>
            </div>
        )
    }
}
