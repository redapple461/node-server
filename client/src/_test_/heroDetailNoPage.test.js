import React from 'react';
import {HeroDetailPage} from '../pages/heroDetailPage'
import { mount,shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Switch,Route } from 'react-router-dom';
import {BrowserRouter as Router} from 'react-router-dom'
import * as actions from '../actions'
import "babel-polyfill"
import jest from 'jest-mock';


describe('<HeroList/> test', () => {
    let store;
    let container;
    const mockStore = configureStore([]);

    // fetch = jest.fn(() => Promise.resolve());

    beforeEach(() => {
        store = mockStore({
            searchUniverse: "",
            heroes: [],
            oldName: "",
            addHero: {name: "", universe: ""},
            detailHero: null,
            isLoad: false,
            noHeroes: true
        });
        store.dispatch = jest.fn();
        
        container = renderer.create(
            <Provider store={store}>
              <Router>
                <Route>
                    <HeroDetailPage match={{params: 'no_hero'}}/>
                </Route>
              </Router>
            </Provider>    
        );

        
    });
        
    

    it('should render with given state from Redux store', done => {
        expect(container.toJSON()).toMatchSnapshot();
        done();
    });


})