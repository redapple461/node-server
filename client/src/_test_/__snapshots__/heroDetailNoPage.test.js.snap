// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroList/> test should render with given state from Redux store 1`] = `
<div>
  <h1>
    Didnt find hero with name : 
    <span
      className="error"
    />
  </h1>
  <a
    href="/main"
    onClick={[Function]}
  >
    <button
      className="waves-effect waves-light btn"
    >
       
      Go to main
       
    </button>
  </a>
</div>
`;
